kvs/include/utils/server_utils.hpp

```diff
@@ -100,6 +101,38 @@ class MemorySetSerializer : public Serializer {
   void remove(const Key& key) { kvs_->remove(key); }
 };
 
+class MemoryCausalSerializer : public Serializer {
+  MemoryCausalKVS* kvs_;
+
+ public:
+  MemoryCausalSerializer(MemoryCausalKVS* kvs) : kvs_(kvs) {}
+
+  string get(const Key& key, unsigned& err_number) {
+    auto val = kvs_->get(key, err_number);
+    if (val.reveal().value.size().reveal() == 0) {
+      err_number = 1;
+    }
+    return serialize(val);
+  }
+
+  unsigned put(const Key& key, const string& serialized) {
+    CausalValue causal_value;
```

Comment:
Shouldn't this just be a `deserialize_causal` method in `common.hpp`?

Reply:
addressed